device.c

```diff
@@ -650,6 +665,9 @@ ssize_t iio_device_debug_attr_read(const struct iio_device *dev,
 ssize_t iio_device_debug_attr_write_raw(const struct iio_device *dev,
 		const char *attr, const void *src, size_t len)
 {
+	if (!attr)
+		return -EINVAL;
```

Comment:
nit: It looks like these little additions have not much to do with this specific patch? Maybe it would make sense to be in a separate patch. They are simple enough though, so fine by me...

Reply:
@nunojsa they actually do have to see with the patch - all the attribute read/write did support bulk read/write of all attributes, which was activated when attr == NULL, and return everything into one giant buffer.